src/apis/storage_devicetree.js

```diff
@@ -20,14 +20,30 @@ import { StorageClient } from "./storage.js";
 
 const INTERFACE_NAME_VIEWER = "org.fedoraproject.Anaconda.Modules.Storage.DeviceTree.Viewer";
 const INTERFACE_NAME_HANDLER = "org.fedoraproject.Anaconda.Modules.Storage.DeviceTree.Handler";
+const INTERFACE_NAME_RESIZABLE = "org.fedoraproject.Anaconda.Modules.Storage.DeviceTree.Resizable";
 const OBJECT_PATH = "/org/fedoraproject/Anaconda/Modules/Storage/DeviceTree";
 
-const callViewer = (...args) => {
-    return _callClient(StorageClient, OBJECT_PATH, INTERFACE_NAME_VIEWER, ...args);
-};
-const callHandler = (...args) => {
-    return _callClient(StorageClient, OBJECT_PATH, INTERFACE_NAME_HANDLER, ...args);
-};
+export class DeviceTree {
+    constructor (deviceTree = OBJECT_PATH) {
+        if (DeviceTree.instance && DeviceTree.instance.deviceTree === deviceTree) {
+            return DeviceTree.instance;
+        }
+        DeviceTree.instance = this;
```

Comment:
I can't figure out what is the role of instance here...

Reply:
NOt needed - not sure what I had in mind when I wrote the class, removing. thanks :)